### PR TITLE
Fix a panic in updater (if update errors and freq = 0)

### PR DESCRIPTION
**Pull request by michelle192837**
This generally only happens when running locally (otherwise freq/--wait is set to something reasonable), and only when there's an error in the update, but it is annoying when it happens.

### CODE REVIEW

File: pkg/updater/updater.go [hunk 1]
```diff
@@ -460,7 +460,11 @@ func Update(parent context.Context, client gcs.ConditionalClient, mets *Metrics,
 		}
 		unprocessed, err := update(ctx, client, log, tg, *tgp, updateGroup, write, gen, fin)
 		if err != nil {
-			delay := freq/4 + time.Duration(rand.Int63n(int64(freq/4)))
+			// delay := freq/4 + time.Duration(rand.Int63n(int64(freq/4)))
```
Comment (chases2) [inline pkg/updater/updater.go:463]: Please delete this comment; the code isn't much use if it's panicky

**Review by chases2 — APPROVED** [review]
/lgtm
/hold for nits

File: pkg/updater/updater.go [hunk 2]
```diff
@@ -460,7 +460,11 @@ func Update(parent context.Context, client gcs.ConditionalClient, mets *Metrics,
...
+			var delay time.Duration
+			if freq > 0 {
```
Comment (fejta) [inline pkg/updater/updater.go:465]: Please don't call Fix with a delay of zero -- meaning retry this error right away, flooding the system with these errors.

Something more like:

```golang
log := log.WithError(err) // shadow
if freq > 0 {
  dleay = blah
  log = log.WithField("delay", delay) // maybe delay.Seconds()?
  q.Fix(tg.Name, etc)
}
log.Error("error updating group)
```

File: pkg/updater/updater.go [hunk 3]
```diff
@@ -460,9 +460,14 @@ func Update(parent context.Context, client gcs.ConditionalClient, mets *Metrics,
...
 		unprocessed, err := update(ctx, client, log, tg, *tgp, updateGroup, write, gen, fin)
 		if err != nil {
-			delay := freq/4 + time.Duration(rand.Int63n(int64(freq/4)))
-			log.WithError(err).WithField("delay", delay).Error("Error updating group")
-			q.Fix(tg.Name, time.Now().Add(delay), true)
+			log := log.WithError(err)
+			var delay time.Duration
+			if freq > 0 {
+				delay = freq/4 + time.Duration(rand.Int63n(int64(freq/4))) // Int63n() panics if freq <= 0
+				log.WithField("delay", delay.Seconds())
```
Comment (fejta) [inline pkg/updater/updater.go:467]: Currently this is a noop

log = log.WithField(...)

**Comment by michelle192837** [comment]
Thanks folks!
/hold cancel

**Review by fejta — APPROVED** [review]
/hold

**Review by fejta — APPROVED** [review]

File: pkg/updater/updater.go (hunk 3 above)
Comment (michelle192837) [reply pkg/updater/updater.go]: Whoops, done.

**Pull request merged**